sdk/ai/azure-ai-inference/src/main/java/com/azure/ai/inference/ChatCompletionClientTracer.java

```diff
@@ -0,0 +1,648 @@
+import com.azure.ai.inference.models.ChatRequestMessage;
+import com.azure.ai.inference.models.ChatRole;
+import com.azure.ai.inference.models.CompletionsFinishReason;
+import com.azure.ai.inference.models.CompletionsUsage;
+import com.azure.ai.inference.models.StreamingChatChoiceUpdate;
+import com.azure.ai.inference.models.StreamingChatCompletionsUpdate;
+import com.azure.ai.inference.models.StreamingChatResponseMessageUpdate;
+import com.azure.ai.inference.models.StreamingChatResponseToolCallUpdate;
+import com.azure.core.http.rest.RequestOptions;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.ClientOptions;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.ConfigurationProperty;
+import com.azure.core.util.ConfigurationPropertyBuilder;
+import com.azure.core.util.Context;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.LibraryTelemetryOptions;
+import com.azure.core.util.TracingOptions;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.tracing.SpanKind;
+import com.azure.core.util.tracing.StartSpanOptions;
+import com.azure.core.util.tracing.Tracer;
+import com.azure.core.util.tracing.TracerProvider;
+import com.azure.json.JsonProviders;
+import com.azure.json.JsonWriter;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayDeque;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.StringJoiner;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/**
+ * Tracing for the convenience methods in {@link com.azure.ai.inference.ChatCompletionsClient} and
+ * {@link com.azure.ai.inference.ChatCompletionsAsyncClient}.
+ * <p>
+ * For more about the OTel semantic conventions this type enables, see
+ * <a href="https://github.com/open-telemetry/semantic-conventions/blob/v1.29.0/docs/gen-ai/azure-ai-inference.md">Azure AI Inference semantic conventions</a>.
+ * </p>
+ */
+final class ChatCompletionClientTracer {
+    private static final String OTEL_SCHEMA_URL = "https://opentelemetry.io/schemas/1.29.0";
+    private static final String INFERENCE_GEN_AI_SYSTEM_NAME = "az.ai.inference";
+    private static final String FINISH_REASON_ERROR = "{\"finish_reason\": \"error\"}";
+    private static final String FINISH_REASON_CANCELLED = "{\"finish_reason\": \"cancelled\"}";
+    private static final StartSpanOptions START_SPAN_OPTIONS = new StartSpanOptions(SpanKind.CLIENT);
+
+    private static final ConfigurationProperty<Boolean> CAPTURE_MESSAGE_CONTENT
+        = ConfigurationPropertyBuilder.ofBoolean("otel.instrumentation.genai.capture_message_content")
```

Comment:
we should use the same env var (and derived config prop name) as we use for python and JS SDKs - `AZURE_TRACING_GEN_AI_CONTENT_RECORDING_ENABLED`